### PR TITLE
[sparkle] - enh: introduce `LinkWrapper`

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR aims at introducing a `LinkWrapper` component to be used by various sparkle components. It is intended to wrap components into a `aLink` component if an "href" prop is passed.

**References:**
- https://github.com/dust-tt/dust/issues/8203

## Risk

Low

## Deploy Plan

Publish `sparkle`

### CODE REVIEW

File: sparkle/src/components/NavigationList.tsx [hunk 1]
```diff
@@ -43,62 +48,94 @@ const NavigationList = React.forwardRef<
...
 NavigationList.displayName = "NavigationList";
 
-interface NavigationListItemProps extends React.HTMLAttributes<HTMLDivElement> {
+interface NavigationListItemProps
+  extends React.HTMLAttributes<HTMLDivElement>,
+    Omit<LinkWrapperProps, "children" | "className"> {
   selected?: boolean;
   label?: string;
   icon?: React.ComponentType;
+  showMoreIcon?: boolean;
```
Comment (JulesBelveze) [inline sparkle/src/components/NavigationList.tsx:57]: Note: shipping this change with this PR even though not related

File: sparkle/src/components/Tabs.tsx [hunk 2]
```diff
@@ -49,7 +69,7 @@ const TabsContent = React.forwardRef<
   <TabsPrimitive.Content
     ref={ref}
     className={cn(
-      "focus-visible:s-ring-ring s-contents s-ring-offset-background focus-visible:s-outline-none focus-visible:s-ring-2 focus-visible:s-ring-offset-2",
+      "s-contents s-ring-offset-background focus-visible:s-outline-none focus-visible:s-ring-2 focus-visible:s-ring-ring focus-visible:s-ring-offset-2",
```
Comment (flvndvd) [inline sparkle/src/components/Tabs.tsx:72]: `s-ring-ring` 🤔

File: sparkle/src/components/Tabs.tsx [hunk 3]
```diff
@@ -27,20 +27,40 @@ const TabsTrigger = React.forwardRef<
...
+        )}
+        {...props}
+        asChild={!!href}
+      >
+        <Button variant="ghost" size="sm" label={label} icon={icon} />
+      </TabsPrimitive.Trigger>
+    );
+
+    return (
+      <LinkWrapper
```
Comment (flvndvd) [inline sparkle/src/components/Tabs.tsx:51]: Looks tempting but won't this add one more layer/fragment in the Tree?

File: sparkle/src/components/NewDropdown.tsx [hunk 4]
```diff
@@ -170,38 +170,70 @@ const NewDropdownMenuContent = React.forwardRef<
...
+    const content = (
+      <DropdownMenuPrimitive.Item
+        ref={ref}
+        className={cn(
+          menuStyleClasses.item({ variant }),
+          inset ? menuStyleClasses.inset : "",
+          className
+        )}
+        {...props}
+        asChild={!!href && asChild}
```
Comment (flvndvd) [inline sparkle/src/components/NewDropdown.tsx:213]: Can we reflect this limitation in the props with a proper exclusion type?

**Review by flvndvd — APPROVED** [review]
LGTM

File: sparkle/src/components/NewDropdown.tsx (hunk 4 above)
Comment (JulesBelveze) [reply sparkle/src/components/NewDropdown.tsx:213]: Well we actually need a `asChild={!!href || asChild}` instead

File: sparkle/src/components/Tabs.tsx (hunk 3 above)
Comment (JulesBelveze) [reply sparkle/src/components/Tabs.tsx:51]: It does indeed, but not sure we have other options here?

Comment (flvndvd) [reply sparkle/src/components/Tabs.tsx:53]: Let's keep it like this for now then.

**Pull request merged**